core/core.go

```diff
@@ -59,16 +59,17 @@ func (c *Core) InsertChain(blocks types.Blocks) (int, error) {
 	nodeCtx := common.NodeLocation.Context()
 	domWait := false
 	for i, block := range blocks {
-		isCoincident := c.sl.engine.IsDomCoincident(block.Header())
 
-		// Write the block body to the db.
-		rawdb.WritePendingBlockBody(c.sl.sliceDb, block.Header().Root(), block.Body())
+		isCoincident := c.sl.engine.IsDomCoincident(block.Header())
+		// Write the block body to the lru cache in worker.
+		rawdb.WriteBody(c.sl.sliceDb, block.Hash(), block.Header().NumberU64(), block.Body())
 
 		if !isCoincident && !domWait {
 			newPendingEtxs, err := c.sl.Append(block.Header(), types.EmptyHeader(), common.Hash{}, big.NewInt(0), false, true, nil)
 			if err != nil {
 				if err == consensus.ErrFutureBlock ||
 					err.Error() == ErrBodyNotFound.Error() ||
+					err.Error() == consensus.ErrPrunedAncestor.Error() ||
```

Comment:
Can you explain this new error case?

Reply:
In validateBody we return this error if we cannot find the state of the parent but have the parent.